sovrin_client/agent/agent.py

```diff
@@ -190,9 +196,18 @@ def __init__(self,
                  attrRepo=None):
 
         Agent.__init__(self, name, basedirpath, client, port, loop=loop)
-        self._wallet = wallet or Wallet(name)
+
+        self.config = getConfig(basedirpath)
+
+        self._restoreLastActiveWallet()
```

Comment:
Let's say the user of this class did not pass any wallet, and we also did not have any previously stored wallet. There is no place where we are instantiaing new wallet instance.